### PR TITLE
Set Selenium Chrome binary and driver path in .env config

### PR DESCRIPTION
**Pull request by primaryobjects**
### Background
- The path to chrome.exe and the Selenium chromedriver may exist in different locations (Linux, Mac, Windows). This change allows setting the path in the `.env` instead of hard-coding the path.

### Changes
- Added config to set Selenium Chrome binary and driver path in `.env`

### Documentation
- Documented within `.env.template`.

```
## CHROMIUM_BINARY_LOCATION - Path to chrome.exe for use with selenium (Optional).
## CHROMIUM_DRIVER_PATH - Path to chrome driver. See https://chromedriver.chromium.org/downloads (default: /usr/bin/chromedriver)
```

### Test Plan
- Unit tested and ran locally successfully.

### PR Quality Checklist
- [X] My pull request is atomic and focuses on a single change.
- [X] I have thoroughly tested my changes with multiple different prompts.
- [X] I have considered potential risks and mitigations for my changes.
- [X] I have documented my changes clearly and comprehensively.
- [X] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Comment by p-i-** [comment]
This is a mass message from the AutoGPT core team.
Our apologies for the ongoing delay in processing PRs.
This is because we are re-architecting the AutoGPT core!

For more details (and for infor on joining our Discord), please refer to:
  https://github.com/Significant-Gravitas/Auto-GPT/wiki/Architecting

**Comment by Boostrix** [comment]
Note this looks like a useful change to be reviewed/integrated despite the ongoing re-arch effort, it's primarily config-level, with very few code changes (and those primarily are about removing hard-coded stuff using best practices like a config dict instead), also it seems to come with testing. So definitely do consider for review/integration to help close some PRs. Thanks

**Comment by jimmylegendary** [comment]
I want to test this pr

File: tests/test_selenium.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    def test_selenium_chromium_binary_location(self):
+        self.config.chromium_binary_location = "custom_chrome.exe"
+
+        with self.assertRaises(WebDriverException) as excep:
+            result = scrape_text_with_selenium("http://localhost")
+
+        self.assertTrue(
+            "no chrome binary at " + self.config.chromium_binary_location
+            in str(excep.exception)
```
Comment (jimmylegendary) [inline tests/test_selenium.py:35]: With this code, test is passed if "no chrome binary" is found at str(excep.execution).
So this should be modified.

self.assertTrue(
            "no chrome binary at " + self.config.chromium_binary_location
            **not in str(excep.exception)**
)

**Comment by jimmylegendary** [comment]
As I tested this PR, there is no problem except test_selenium.py as I commented


But, for next task,
we might need to not use **executable_path** because this parameter is deprecated after selenium4.
We can use Service object in place of using executable_path.

File: tests/test_selenium.py (hunk 1 above)
Comment (primaryobjects) [reply tests/test_selenium.py:35]: Thank you for the review! The test is correctly confirming the message, as this is a negative test case, where the exception is confirmed to raise. We expect to get back "no chrome binary".

**Comment by Pwuts** [comment]
Thanks for submitting! It was a challenge getting browsing to work reliably across multiple platforms, and I wish we had picked up on this PR earlier. By now, it has been addressed in a slightly different way afaics:
https://github.com/Significant-Gravitas/Auto-GPT/blob/29c2476e3090aef8afa56c682f31778143864af5/autogpt/commands/web_selenium.py#L140-L145

I'll look if we can still integrate the tests you wrote. :)

**Comment by primaryobjects** [comment]
I'm ok with closing this PR and deferring it to a later time, if needed.

Note, the proposed alternative still appears to [hardcode](https://github.com/Significant-Gravitas/Auto-GPT/blob/7c4fc45b4a536e6423b6491c34953a2516863d78/autogpt/commands/web_selenium.py#L132) the path to chromedriver. It also does not allow configuring a custom path to the chrome.exe binary.

This PR addresses both issues by allowing the user to customize the path locations - notably on Windows, where the path to the driver and executable may differ.

It's ok either way, we have the PR and can always reopen if needed. Thank you.

**Comment by Pwuts** [comment]
> Note, the proposed alternative still appears to [hardcode](https://github.com/Significant-Gravitas/Auto-GPT/blob/7c4fc45b4a536e6423b6491c34953a2516863d78/autogpt/commands/web_selenium.py#L132) the path to chromedriver.

Yup, because that's the location of the driver in the release docker image and on Linux. On Windows, it will not be in that location, and the program will manage by downloading a driver on the fly. This is a much simpler solution than making everything configurable, hence the proposal to close this PR. Unless of course the current implementation causes problems somehow.

> It also does not allow configuring a custom path to the chrome.exe binary.

What would be the use case for such a config option?

**Pull request closed without merging**